olp-cpp-sdk-core/src/client/HRN.cpp

```diff
@@ -51,62 +51,73 @@ std::string HRN::ToString() const {
 }
 
 std::string HRN::ToCatalogHRNString() const {
-  if (this->service != ServiceType::Data) return std::string();
+  if (service != ServiceType::Data) return std::string();
 
   return "hrn:" + partition + ":data:" + region + ":" + account + ":" +
-         this->catalogId;
+         catalogId;
 }
 
 HRN::HRN(const std::string& input) {
   Tokenizer tokenizer(input, ':');
 
   // hrn must start with "hrn:"
-  if (!tokenizer.hasNext()) return;
+  if (!tokenizer.HasNext()) return;
 
-  std::string protocol = tokenizer.next();
-
-  // as an exception, also allow passing in direct URLs. Eases the use of
-  // command line tools that want to access custom catalog URLs.
-  if (protocol == "http" || protocol == "https") {
-    this->catalogId = input;
-    this->partition = "catalog-url";
-    return;
-  }
+  std::string protocol = tokenizer.Next();
 
   if (protocol != "hrn") return;
 
   // fill up the rest of the fields
-  if (tokenizer.hasNext()) this->partition = tokenizer.next();
-  if (tokenizer.hasNext()) {
-    auto serviceStr = tokenizer.next();
-    if (serviceStr.compare("data") == 0)
-      this->service = ServiceType::Data;
-    else if (serviceStr.compare("schema") == 0)
-      this->service = ServiceType::Schema;
-    else if (serviceStr.compare("pipeline") == 0)
-      this->service = ServiceType::Pipeline;
-    else {
-      this->service = ServiceType::Unknown;
+  if (tokenizer.HasNext()) {
+      partition = tokenizer.Next();
+  }
+
+  if (tokenizer.HasNext()) {
+    auto serviceStr = tokenizer.Next();
```

Comment:
```suggestion
    auto service_str = tokenizer.Next();
```
Please, change the naming of variables in this method according to Google C++ style